lib/queries.js

```diff
@@ -171,16 +242,297 @@ export async function getAllOvoAndKboCouples() {
+                                        )} ;
+                                        <http://mu.semte.ch/vocabularies/ext/rechtsvorm> ${sparqlEscapeString(
+                                          kboOrg.rechtsvorm
+                                        )} ;
+                                        <http://mu.semte.ch/vocabularies/ext/shortName> ${sparqlEscapeString(
+                                          kboOrg.shortName
+                                        )} ;
+                                        <http://mu.semte.ch/vocabularies/ext/startDate> ${sparqlEscapeString(
+                                          kboOrg.startDate
+                                        )} ;
+                                        <http://www.w3.org/2004/02/skos/core#prefLabel> ${sparqlEscapeString(
+                                          kboOrg.organisationName
+                                        )} ;
+                                        <http://www.w3.org/2004/02/skos/core#altLabel> ${sparqlEscapeString(
+                                          kboOrg.formalName
+                                        )} ;
+                                        <https://data.lblod.info/vocabularies/generiek/geplandeEindDatum> ${sparqlEscapeString(
+                                          kboOrg.changeTime
+                                        )} ;
+                                        <http://www.w3.org/ns/adms#identifier> ${sparqlEscapeUri(
+                                          kboId
+                                        )} ;
+                                        <http://www.w3.org/ns/regorg#orgStatus> ${sparqlEscapeUri(
+                                          kboOrg.activeState
+                                        )} ;
+                                        <http://schema.org/contactPoint> ${sparqlEscapeUri(
+                                          contactPointUri
+                                        )} .
+    }
+  }WHERE{
+    VALUES ?g {
+      <http://mu.semte.ch/graphs/administrative-unit>
+    }
+  }   
+  `;
+
+  await update(updateStr);
+  return newKboIdUri;
+}
+
+export async function updateKboOrg(newKboInfo, identifiers) {
+  //update Adress
+  let updateStr = `
+  DELETE{
+    GRAPH ?g{
+      ?adressUri a <http://www.w3.org/ns/locn#Address> ;
+                  <http://www.w3.org/ns/locn#fullAddress> ?fulladdress  .
+    }
+  }
+  INSERT{
+    graph ?g{ 
+      ?adressUri a <http://www.w3.org/ns/locn#Address> ;
+                <http://www.w3.org/ns/locn#fullAddress> ${sparqlEscapeString(newKboInfo.formattedAddress)} .
+    }
+  }
+  WHERE{
+    VALUES ?g {
+      <http://mu.semte.ch/graphs/administrative-unit> 
+    }
+    GRAPH ?g {
+      ?adressUri a <http://www.w3.org/ns/locn#Address> ;
+                <http://www.w3.org/ns/locn#fullAddress> ?fulladdress  .
+      
+      FILTER(?adressUri = ${sparqlEscapeUri(identifiers.addressUri)})
+    }
+  }  
+  `;
```

Comment:
The query will only run when newKboInfo.changeTime > indentifiers.changeTime but it will return be false anyhow if either of those is not defined. So i do not think this will cause a problem. 

For the formattedAdress i can indeed do the extra check indeed